src/components/NFImage/NFImage.tsx

```diff
@@ -0,0 +1,99 @@
+import styles from "./NFImage.module.sass";
+
+import React, { useState } from "react";
+
+interface Properties {
+  owner: string;
+  title: string;
+  imageUrl: string;
+  avatarUrl?: string;
+  name?: string;
+}
+
+function NFImage(props: Properties) {
+  const { owner, title, imageUrl, avatarUrl, name } = props;
+
+  const [modalWindowIsOpen, setModalWindowIsOpen] = useState(false);
+
+  const openModalWindow = () => setModalWindowIsOpen(true);
+  const closeModalWindow = () => setModalWindowIsOpen(false);
+
+  function getLimitedString(text: string, maxlength: number, lastSymbols: number, isBeautifulEnding: boolean = true, textMissingWarning?: string): string {
+    if (text.length === 0 && textMissingWarning) {
+      return textMissingWarning;
+    }
+
+    if (text.length <= maxlength) {
+      return text;
+    }
+
+    if (lastSymbols) {
+      return `${text.substring(0, maxlength)}...${text.substring(text.length - lastSymbols)}`;
+    }
+
+    let limitedString = text.substring(0, maxlength)
+
+    if (isBeautifulEnding && !lastSymbols) {
+      let lastSymbolsAfterSpace = limitedString.split(" ").slice(-1)[0];
+
+      const isManyLastSymbols = (
+        lastSymbolsAfterSpace.length === limitedString.length || 
+        lastSymbolsAfterSpace.length >= limitedString.length/2
+      );
+      if (isManyLastSymbols) {
+        return `${limitedString}...`;
+      }
+
+      const lengthLimitedString = limitedString.length;
+      const lengthLastSymbolsAndSpace = lastSymbolsAfterSpace.length+1;
+      const maxLengthForBeautifulText = lengthLimitedString - lengthLastSymbolsAndSpace;
+      
+      const limitedStringWithoutLastSymbols = limitedString.substring(0, maxLengthForBeautifulText);
+      return `${limitedStringWithoutLastSymbols}...`;
+    }
+
+    return `${limitedString}...`;
+  }
+
```

Comment:
mb it's better to make some kind of module with functions - utilities for such functions?
I guess we could be using this function in another components right?